MidProject Review PR (v2)

naf445 opened the pull request:
Mid-Project Review Pull Request (Version 2)

I initially made a PR but accidentally closed and merged it myself so this is a second PR.

New additions in this branch:

Updated README with detailed information on how to run new scripts committed during development of this branch.
Bash script getData.sh which allows for getting data in raw form from source and directing to S3 bucket
Bash script mysqlconfig.sh which allows for setting up MySQL information before connection to RDS
Python script sqlalchemy_mysql_setup.py which allows (using command line arguments) for the setting up of a schema and database instance either on RDS or locally using SQLite.
Directory reviewDeck containing reveal.js presentation slides for mid-project review assignment

Commit:
make small change so you can open pull request

tonycolucci commented:
Initial attempt to pull and recreate (README):
I had to enter "bash getData.sh <s3 repo name>", so consider changing that instruction or at least suggesting the alternative syntax.

Commit:
Fix Readme to reflect bash command instructions during setup

tonycolucci commented:
Traceback (most recent call last):
  File "sqlalchemy_mysql_setup.py", line 7, in <module>
    import pandas as pd
  File "C:\Users\tonyc\Anaconda3\lib\site-packages\pandas\__init__.py", line 19, in <module>
    "Missing required dependencies {0}".format(missing_dependencies))
ImportError: Missing required dependencies ['numpy']

I'm getting the above error when running python sqlalchemy_mysql_setup.py RDS, but I'm pretty sure it's an issue with my Anaconda install. Should I be setting up a conda environment with certain packages in order to run this?

tonycolucci commented:
For posterity, this is a QA pass over Nathan's mid-project repository for his AVC project app. My plan is to first clone the repo to my local machine and run through the instructions in his readme, then do a code review and leave some overall comments.

tonycolucci commented:
Created a virtualenv and attempted to run requirements.txt, but encountered the error:
Collecting mkl-fft==1.0.10 (from -r requirements.txt (line 32))
  ERROR: Could not find a version that satisfies the requirement mkl-fft==1.0.10 (from -r requirements.txt (line 32)) (from versions: none)
ERROR: No matching distribution found for mkl-fft==1.0.10 (from -r requirements.txt (line 32))

Looking at https://pypi.org/project/mkl-fft/#history I'm seeing versions 1.0.0.17, 1.0.2 and 1.0.6. For now I'm going to try changing this to mkl-fft>=1.0.10 and see if it works.

UPDATE: I've tried several version for both mkl-fft and mkl_fft, but I can't find a version that installs. I'm going to remove this from requirements.txt for now and see how far I can get.
UPDATE #2: The same thing happened for mkl-random==1.0.13, so I moved both of those to the bottom of requirements.txt and everything else installed.

Commit:
Reorganize repository and update README to reflect changes.

This refactoring moved files around, though I haven't gone through all the noteboooks and scripts to update the references from the files, so that will be a next step. I tried to place config files and scripts in their respective correct directories and get the organization of my repository more sensible.

tonycolucci commented:
Everything after then worked well, nice job and good instructions!
One nitpick, the sqlite instructions suggested that they would create "msia423.db", but it actually creates "avc423.db", so that should be updated.

Commit:
Add section in README about environment setup (virtual-env or conda env)

Because other developers were not explicitly instructed to activate either a conda environment or a virtual environment, they have had trouble running the present python or bash scripts. Thus the README now reflects these explicit instructions.

Commit:
Update README to reflect the correct name of the database created.

Update sqlalchemy setup python script so that both databases created either locally or on AWS RDS will have the same name, and the name referenced in the README

tonycolucci commented:
In sqlalchemy_mysql_setup.py, I noticed you're setting string lengths of 1000 for user input and 50 for model output. These likely seem reasonable, but they may be something to consider adding to a configuration so you can make sure these limits are reflected throughout your application.

tonycolucci commented:
The review deck looks great! I think you highlight a lot of great work so far. Two issues:
1. Under Demo/Analysis/Initial LDA Run Without Name Filtering there is a broken image link. 
2. The README under the reviewDeck folder doesn't seem to make sense. I would suggest either removing it, or clarifying its use.

tonycolucci commented:
Overall QA Review:
### Functionality
The functions in the code work well for what they are meant to do. The ability to create a database in RDS or Sqlite was a nice feature. The only issues with functionality were in the requirements file.
### Readability and code quality
sqlalchemy_mysql_setup.py is slightly difficult to read in sections where comments are interspersed with lines of code that have been commented out without any lines in between. Removing unused pieces of code, adding comments and whitespace will improve this.
Design decisions are easy to follow, I can tell what data is contained in variables easily, and I don't see anything that is really needing modularization.
### Overall design
I can see how this code fits into the larger project. There are certain aspects of the code that could be improved with an eye on extensibility. Above, I mentioned the size of the strings for user input and model output. An eye on environment setup may help a bit, but I don't see any major concerns as these improvements could be made relatively easily at this point in the project.
### Testing
The code is structured in a way that could be tested. It doesn't currently have logging or testing setup.
### Documentation
The documentation overall is highly effective. The README does an excellent job of walking me through the steps to recreate the work so far. The pull request is clear about what has been updated. Each commit statement is concise and gives an idea of what was updated. The documentation in the code is relatively sparse at this point, though since the codebase is not very large or complex to this point, it is still not too difficult to follow. I would suggest getting on top of adding docstrings and argument descriptions to existing functions before you need to come back to them a couple weeks from now and quickly understand how they are feeding into the new functions building off of them. Overall excellent work!

Commit:
Add logging capabilities, restructure repo, move notebook work to script.

Added logging config and some test logs into a couple of scripts. Moved some notebooks into demo folder. Cleaned out reviewdeck folder. Got rid of scripts not being used yet. Transform custom transformer notebook into a script which can be called with command line arguments and uses an sklearn pipeline to clean your book summary data.

and some test logging into various scripts.
